packages/teleterm/src/ui/TopBar/NavigationMenu/NavigationItem.tsx

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import styled from 'styled-components';
+
+import { Box, Text, Flex } from 'design';
+
+import { AccessRequestDocumentState } from 'teleterm/ui/services/workspacesService';
+import { useAppContext } from 'teleterm/ui/appContextProvider';
+
+export function NavigationItem({
+  Icon,
+  title,
+  state,
+  closeMenu,
+  clusterUri,
+}: NavigationItemProps) {
+  const ctx = useAppContext();
+  const docService = ctx.workspacesService.getActiveWorkspaceDocumentService();
+
+  const onClick = () => {
+    if (!docService) {
+      return;
+    }
+    const doc = docService.createAccessRequestDocument({ clusterUri, state });
+    docService.add(doc);
+    docService.open(doc.uri);
```

Comment:
This logic should live in `navigationItems`: 
```typescript
{
   title: 'New Access Request',
   Icon: <Add fontSize={2} />
   onNavigate: () => {...}
},
```

`NavigationItem` doesn't need to know about `docService` or anything like that.

Reply:
I agree, because we may need to do something special per item eventually. How is `navigationItems` supposed to be aware of the doc service? I'm a bit lost there, can you point to an existing pattern similar to this?

Reply:
This is awesome thank you. yes, absolutely the best path to go with.

Reply:
This came out great! Wonderful suggestion